src/Commands/MakeFactoryCommand.php

```diff
@@ -0,0 +1,128 @@
+<?php
+
+namespace KoalaFacade\DiamondConsole\Commands;
+
+use Illuminate\Console\Command;
+use Illuminate\Contracts\Filesystem\FileNotFoundException;
+use Illuminate\Support\Str;
+use KoalaFacade\DiamondConsole\Actions\Filesystem\FilePresentAction;
+use KoalaFacade\DiamondConsole\Actions\Stub\CopyStubAction;
+use KoalaFacade\DiamondConsole\Commands\Concerns\HasArguments;
+use KoalaFacade\DiamondConsole\Commands\Concerns\HasOptions;
+use KoalaFacade\DiamondConsole\Commands\Concerns\InteractsWithPath;
+use KoalaFacade\DiamondConsole\DataTransferObjects\CopyStubData;
+use KoalaFacade\DiamondConsole\DataTransferObjects\Filesystem\FilePresentData;
+use KoalaFacade\DiamondConsole\DataTransferObjects\PlaceholderData;
+
+class MakeFactoryCommand extends Command
+{
+    use InteractsWithPath, HasArguments, HasOptions;
+
+    protected $signature = 'diamond:factory {name} {domain} {--force}';
+
+    protected $description = 'Create model Factory';
+
+    /**
+     * @throws FileNotFoundException
+     */
+    public function handle(): void
+    {
+        $this->info(string: 'Generating factory & interface file to your project');
+
+        $this->resolveFactoryContract();
+
+        $this->resolveFactoryConcrete();
+    }
+
+    /**
+     * @throws FileNotFoundException
+     */
+    protected function resolveFactoryContract(): void
+    {
+        $fileName = $this->resolveNameArgument() . 'Contract' . '.php';
+
+        $namespace = $this->resolveNamespace(
+            identifier: 'Models\\Contracts\\',
```

Comment:
i still not sure wha this one generate for?

Reply:
i was expect this factory contract will generate into Domain/Shared/Contracts/Database/Factories namespace, wdyt about this one

Reply:
binding at service provider, i want to reduce navigation issue.

Reply:
yes i mean the namespace we talk about

Reply:
i want make it a part of shared/Domain/Live here.

Reply:
what if Domain/Shared/InputedDomain/Factory ?

Reply:
yep but not satisfied for me. coz it shouldn't tied up with models directly. any input?

Reply:
i was comment this previously, https://github.com/KoalaFacade/diamond-console/pull/26/files/07d3e91897e57238167472120d395129df60f926#r1038959056

Reply:
oke then how bout if there so many interface?

Reply:
good question

Reply:
<img width="261" alt="image" src="https://user-images.githubusercontent.com/56016477/205496567-871a2fd1-9537-4e27-a79b-b55c38d2131c.png">

this is how its will looks like

Reply:
aite